pkg/vulnsrc/azure/azure.go

```diff
@@ -199,30 +203,31 @@ func resolveTests(dir string) (map[string]resolvedTest, error) {
 }
 
 func followTestRefs(test oval.RpmInfoTest, objects map[string]string, states map[string]oval.RpmInfoState) (resolvedTest, error) {
+	eb := oops.With("object_ref", test.Object.ObjectRef).With("state_ref", test.State.StateRef).With("test_ref", test.ID)
+
 	// Follow object ref
 	if test.Object.ObjectRef == "" {
-		return resolvedTest{}, xerrors.New("invalid test, no object ref")
+		return resolvedTest{}, eb.Errorf("invalid test, no object ref")
 	}
 
 	pkgName, ok := objects[test.Object.ObjectRef]
 	if !ok {
-		return resolvedTest{}, xerrors.Errorf("invalid test data, can't find object ref: %s, test ref: %s",
-			test.Object.ObjectRef, test.ID)
+		return resolvedTest{}, eb.Errorf("invalid test data, can't find object ref")
 	}
 
 	// Follow state ref
 	if test.State.StateRef == "" {
-		return resolvedTest{}, xerrors.New("invalid test, no state ref")
+		return resolvedTest{}, eb.Errorf("invalid test, no state ref")
 	}
 
 	state, ok := states[test.State.StateRef]
 	if !ok {
-		return resolvedTest{}, xerrors.Errorf("invalid tests data, can't find ovalstate ref %s, test ref: %s",
-			test.State.StateRef, test.ID)
+		return resolvedTest{}, eb.Errorf("invalid tests data, can't find ovalstate ref")
 	}
 
+	eb = eb.With("data_type", state.Evr.Datatype)
 	if state.Evr.Datatype != "evr_string" {
-		return resolvedTest{}, xerrors.Errorf("state data type (%s): %w", state.Evr.Datatype, ErrNotSupported)
+		return resolvedTest{}, eb.Wrapf(ErrNotSupported, "state data type")
```

Comment:
nil:
we need `data_type` only for this error
```suggestion
		return resolvedTest{}, eb.With("data_type", state.Evr.Datatype).Wrapf(ErrNotSupported, "state data type")
```